vm/src/slots.rs

```diff
@@ -2,8 +2,28 @@ use crate::function::{OptionalArg, PyFuncArgs, PyNativeFunc};
 use crate::pyobject::{IdProtocol, PyObjectRef, PyRef, PyResult, PyValue};
 use crate::VirtualMachine;
 
+bitflags! {
+    pub struct PyTpFlags: u64 {
+        const DEFAULT = 0;
+        const BASETYPE = 1 << 10;
+    }
+}
+
+impl PyTpFlags {
+    pub fn has_feature(self, flag: Self) -> bool {
+        (self & flag).bits != 0
```

Comment:
I think you could just use the `contains` method that is generated by bitflags instead of defining this method.